packages/api/src/services/parseHearingOutcome.ts

```diff
@@ -0,0 +1,36 @@
+import type { AnnotatedHearingOutcome } from "@moj-bichard7/core/types/AnnotatedHearingOutcome"
+import type { FastifyBaseLogger } from "fastify"
+
+import { isError } from "@moj-bichard7/common/types/Result"
+import parseAhoXml from "@moj-bichard7/core/lib/parse/parseAhoXml/parseAhoXml"
+import parseAnnotatedPncUpdateDatasetXml from "@moj-bichard7/core/phase2/parse/parseAnnotatedPncUpdateDatasetXml/parseAnnotatedPncUpdateDatasetXml"
+
+const isPncUpdateDatasetFromXml = (message: string) => /<AnnotatedPNCUpdateDataset/.exec(message)
+
+const parseHearingOutcome = (hearingOutcome: string, logger?: FastifyBaseLogger): AnnotatedHearingOutcome | Error => {
```

Comment:
I wonder if `logger` should be a required parameter. I would assume we always want to log errors?

Reply:
I think I'd expect parseHearingOutcome to return a result or an error and the handler to do the logging

Reply:
@emadkaramad We don't in tests.

@richardcane We use the logger instead of using `console.error` https://github.com/ministryofjustice/bichard7-next-core/blob/main/packages/ui/src/utils/parseHearingOutcome.ts#L24

Reply:
@emadkaramad https://dsdmoj.atlassian.net/browse/BICAWS7-3311